docs/external-node.md

```diff
@@ -0,0 +1,565 @@
+
+![OVS pipeline](assets/ovs-pipeline-external-node.svg)
+
+### Non-IP packet
+
+`NonIPTable` is a new OpenFlow table introduced only on an external Node,
+which is dedicated for all non-IP packets. A non-IP packet is forwarded between
+the pair ports directly, e.g., a non-IP packet entering OVS from the uplink
+interface is output to the paired internal port, and a packet from the internal
+port is output to the uplink.
+
+### IP packet
+
+A new OpenFlow pipeline is set up on an external Node to process the IP packet.
+Antrea NetworkPolicy enforcement is the major function in this new pipeline, and
+the OpenFlow tables used are similar to the Pod pipeline. **No L3 routing is
+provided on an external Node**, and a simple L2 forwarding policy is
+implemented. OVS connection track is used to assist the NetworkPolicy function,
+in a result only the first packet is validated by the OpenFlow entries, and the
+sequencing packets in an accepted connection are allowed directly.
+
+- Egress/Ingress Tables
+
+Table `XgressSecurityClassifierTable` is installed in both `stageEgressSecurity`
+and `stageIngressSecurity`, which is used to install the OpenFlow entries for
+the [`policyBypassRules`](#bypass-antrea-networkpolicy) in the agent configuration.
+
+This is an example of the OpenFlow entry for the above configuration:
+
+```yaml
+table=IngressSecurityClassifier, priority=200,ct_state=+new+trk,tcp,nw_src=1.1.1.1,tp_dst=22 actions=resubmit(,IngressMetric)
+```
+
+Other OpenFlow tables in `stageEgressSecurity` and `stageIngressSecurity` are
+the same as those installed on a Kubernetes worker Node. For more details about
+these tables, please refer to the general [introduction](design/ovs-pipeline.md)
+of Antrea OVS pipeline.
+
+- L2 Forwarding Tables
+
+`L2ForwardingCalcTable` is used to calculate the expected output port of an IP
+packet. As the pair ports with the internal port and uplink always exist on the
+OVS bridge, and both interfaces are configured with the same MAC address, the
+match condition of an OpenFlow entry in `L2ForwardingCalcTable` uses the input
+port number but not the MAC address of the packet. The flow actions are:
+
+1) set flag `OFPortFoundRegMark`, and
+2) set the peer port as the `TargetOFPortField`, and
+3) enforce the packet to go to stageIngressSecurity.
+
+Below is an example OpenFlow entry in `L2ForwardingCalcTable`
+
+```yaml
+table=L2ForwardingCalc, priority=200,ip,in_port=ens224 actions=load:0x1->NXM_NX_REG0[8],load:0x7->NXM_NX_REG1[],resubmit(,IngressSecurityClassifier)
+table=L2ForwardingCalc, priority=200,ip,in_port="ens224~" actions=load:0x1->NXM_NX_REG0[8],load:0x8->NXM_NX_REG1[],resubmit(,IngressSecurityClassifier)
+```
+
+## Limitations
+
+This feature currently supports only one interface per `ExternalNode` object,
+and `ips` must be set in the interface. The support for multiple network
+interfaces will be added in the future.
+
+`ExternalNode` name must be unique in the `cluster` scope.
```

Comment:
`ExternalNode` name must be unique in the `cluster` scope even though it is itself a Namespaced resource.